### PR TITLE
fix: upgrade to latest react-navigation packages versions

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Bugfix**

This PR resolves the following crash
Screenshot 2021-09-20 at 11 50 24

### Description
The same crash is happening because of react-navigation, it seems like upgrading to a later version fixes it. Let's see because I still couldn't reproduce the bug on the simulator and iPhone.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Upgrade to latest react-navigation versions - mounir

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Upgrade to latest react-navigation versions - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f1c732c962378530023be257ebd3717572f978f7
</p>

File: Podfile.lock [hunk 1]
```diff
@@ -820,14 +820,7 @@ DEPENDENCIES:
...
 SPEC REPOS:
-  https://github.com/artsy/Specs.git:
-    - Aerodramus
-    - "Artsy+UIColors"
-    - "Artsy+UIFonts"
-    - "Artsy+UILabels"
-    - Artsy-UIButtons
-    - Extraction
-  trunk:
+  https://cdn.cocoapods.org/:
```
Comment (brainbicycle) [inline Podfile.lock:823]: I think this is same podfile churn, can we remove?

**Comment by brainbicycle** [comment]
Minor comment on podfile.lock other than that looks good to me!

File: Podfile.lock (hunk 1 above)
Comment (MounirDhahri) [reply Podfile.lock:823]: 😭😭😭 I will

**Pull request merged**